### DISCUSSION TITLE
SD Card / Mac

### DISCUSSION THREAD

**Discussion by volkerj70** [jomjol/AI-on-the-edge-device#3355] [Q&A]
I use a 8GB SD card and tried several once, all failed.

I get this protocol if i restart by console:

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
I (27) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (27) boot: compile time 20:02:47
I (27) boot: chip revision: 3
I (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (46) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (56) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (75) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000
I (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000
I (104) boot: End of partition table
I (109) boot_comm: chip revision: 3, min. application chip revision: 0
I (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=40014h (262164) map
I (225) esp_image: segment 1: paddr=0005003c vaddr=3ffb0000 size=040e8h ( 16616) load
I (232) esp_image: segment 2: paddr=0005412c vaddr=40080000 size=0beech ( 48876) load
I (253) esp_image: segment 3: paddr=00060020 vaddr=400d0020 size=1100f4h (1114356) map
I (680) esp_image: segment 4: paddr=0017011c vaddr=4008beec size=0e354h ( 58196) load
I (718) boot: Loaded app from partition at offset 0x10000
I (718) boot: Disabling RNG early entropy source...
I (730) cpu_start: Multicore app
I (730) quad_psram: This chip is ESP32-D0WD
I (731) esp_psram: Found 8MB PSRAM device
I (731) esp_psram: Speed: 40MHz
I (734) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (741) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (749) cpu_start: Pro cpu up.
I (752) cpu_start: Starting app cpu, entry point is 0x40082e0c
I (0) cpu_start: App cpu up.
I (1636) esp_psram: SPI SRAM memory test OK
I (1645) cpu_start: Pro cpu start user code
I (1645) cpu_start: cpu freq: 160000000 Hz
I (1645) cpu_start: Application information:
I (1646) cpu_start: Project name:     AI-on-the-edge
I (1650) cpu_start: App version:      v15.7.0
I (1654) cpu_start: Compile time:     Feb 17 2024 00:03:24
I (1660) cpu_start: ELF file SHA256:  71b36fda6e4066fa...
I (1665) cpu_start: ESP-IDF:          5.1.2
I (1669) cpu_start: Min chip rev:     v0.0
I (1672) cpu_start: Max chip rev:     v3.99 
I (1676) cpu_start: Chip rev:         v3.1
I (1681) heap_init: Initializing. RAM available for dynamic allocation:
I (1687) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1692) heap_init: At 3FFBC688 len 00023978 (142 KiB): DRAM
I (1697) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1703) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1708) heap_init: At 4009A240 len 00005DC0 (23 KiB): IRAM
I (1714) esp_psram: Adding pool of 4082K of PSRAM memory to heap allocator
I (1722) spi_flash: detected chip: generic
I (1724) spi_flash: flash io: dio
I (1739) app_start: Starting scheduler on CPU0
I (1739) app_start: Starting scheduler on CPU1
I (1739) main_task: Started on CPU0
I (1749) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1749) main_task: Calling app_main()
I (1749) MAIN: 

================ Start app_main =================
E (1799) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
E (1799) sdcard_init: sdmmc_card_init failed (0x107).
E (1799) MAIN: SD card init failed. Check if SD card is properly inserted into SD card slot or try another card
E (1809) MAIN: Device init aborted!
I (1819) main_task: Returned from app_main()

What should I try or do?

**Comment by friedpa** [comment]
In same cases the SD Card Reader is defect, have you tried also another ESP Device?

**Reply by volkerj70** [reply]
No I have only this one.
Can this be an issue: 
W (741) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped

**Reply by Slider0007** [reply]
> No I have only this one.
> Can this be an issue: 
> W (741) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped

This is normal, no issue. ESP32 can only address 4MB of the available 8MB

**Comment by Slider0007** [comment]
There are issues when SD is formatted with a Mac. Just search for multiple discussions here or use Windows / Linux to format SD card.

**Reply by volkerj70** [reply]
I tried to format as well on a Win PC. 
Same behaviour :-(

**Reply by Slider0007** [reply]
This failure 0x107 indicates clearly an issue with SD card or all related parts of SD slot (e.g. bad wiring).

Try another board or another SD card. The device is quite picky with SD cards ..

**Reply by SybexX** [reply]
https://github.com/jomjol/AI-on-the-edge-device/issues/2945#issuecomment-2371500469
